packages/docusaurus-theme-common/src/utils/configUtils.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Tries to create a RegExp from a string, coming from the Docusaurus config.
+ * If it fails to create a RegExp it returns the input string.
+ *
+ * @param possibleRegexp string that is possibly a regex
+ * @returns a Regex if possible, otherwise the string
+ */
+export function getRegexpOrString(possibleRegexp: string): RegExp | string {
```

Comment:
🤔 I don't really like this implementation and would prefer to just support regexp. Adding unit tests would likely reveal the problematic cases.

My idea was more:
- config accepts both string + RegExp
- config validation normalizes this as a serializable RegExp string
- client-side always use `new RexExp()`

The config normalization can be something like:

```
if (from instanceof string) {
  return escapeRegexp(from);
} else if (from instanceof RegExp) {
 return from.source;
} else {
  throw new Error('unexpected")
}
```

=> always return a valid Regexp string source

Does it make sense?

Reply:
That makes sense; I wasn't terribly happy with `getRegexpOrString` either. I've tried to address your suggestions in the latest commit, hope I got it right this time 😅